src/environment.rs

```diff
@@ -0,0 +1,196 @@
+/// Structure to deal with environment variables
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Environment {
+    /// Customized environment variables
+    vars: Vec<(String, String)>,
+    /// Define if the structure must inherit
+    inherit: bool,
+}
+
+impl Default for Environment {
+    fn default() -> Self {
+        Self {
+            vars: vec![],
+            inherit: false,
+        }
+    }
+}
+
+impl Environment {
+    /// Create a new Environment object with inherit
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// let e = assert_cli::Environment::inherit().compile();
+    /// let e_: Vec<(String, String)> = ::std::env::vars().collect();
+    ///
+    /// assert_eq!(e, e_);
+    /// ```
+    pub fn inherit() -> Self {
+        Self {
+            vars: vec![],
+            inherit: true,
+        }
+    }
+
+    /// Create a new Environment object without inheriting
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// let e = assert_cli::Environment::empty().compile();
+    /// assert_eq!(e, Vec::new());
+    /// ```
+    pub fn empty() -> Self {
+        Self::default()
+    }
+
+    /// Insert a new entry into the custom variables for this environment object
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// let e = assert_cli::Environment::empty().insert("foo", "bar").compile();
+    /// assert_eq!(e, vec![("foo".to_string(), "bar".to_string())]);
+    /// ```
+    pub fn insert<S1: Into<String>, S2: Into<String>>(mut self, key: S1, val: S2) -> Self {
+        self.vars.push((key.into(), val.into()));
+        self
+    }
+
+    /// Compile Environment object
+    pub fn compile(&self) -> Vec<(String, String)> {
+        if self.inherit {
+            ::std::env::vars().chain(self.vars.clone()).collect()
+        } else {
+            self.vars.clone()
```

Comment:
Should `Environment` be moved into this?

This would give the caller control on whether the `.clone()`s are needed or not.  In our case, we call `compile` when consuming `Assert`, so we wouldn't need to call any of these `.clone()`s.